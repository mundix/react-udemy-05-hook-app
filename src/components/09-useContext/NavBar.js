import React from 'react';
import { Link, NavLink} from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link className="navbar-brand" to='/'>useContext</Link>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName='active' to='/' className='nav-link'>Home</NavLink>
                    <NavLink exact activeClassName='active' to='/about' className='nav-link '>About</NavLink>
                    <NavLink exact activeClassName='active' to='/login' className='nav-link '>Login</NavLink>
                </div>
            </div>
        </nav>
    )
}
