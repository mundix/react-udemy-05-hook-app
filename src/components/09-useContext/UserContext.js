import {createContext} from 'react';

// Higher order component, como el swith como el router compoentns 
// Eso queire decir que puedo colocar componentes hijos
export const UserContext  = createContext(null);
