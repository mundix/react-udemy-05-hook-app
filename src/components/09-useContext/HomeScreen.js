import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    // Asi es como puedo acceder al user obj mediente UserContext 
    // Me devuelve todo lo que envie en el value del UserContext.Provider
    // const userContext = useContext(UserContext);
    const {user} = useContext(UserContext);
    // console.log(user);

    return (
        <div>
            <h1>Home Screen</h1>
            <hr />
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
        </div>
    )
}
