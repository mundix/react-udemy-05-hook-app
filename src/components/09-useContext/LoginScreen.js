import { getDefaultNormalizer } from '@testing-library/react';
import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    // 1. Obtejer referencia al userContext
    const {setUser} = useContext(UserContext);
    // 2. setUSer
    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button className="btn btn-primary"
                onClick={() => {
                    setUser({
                        id: 234,
                        name: 'John Connors',
                        email: 'connorsj@getDefaultNormalizer.com'
                    })
                }}
            >
                Login
            </button>
        </div>
    )
}
