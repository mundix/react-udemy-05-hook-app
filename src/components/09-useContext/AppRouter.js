import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            {/* Se recomienda usar el div  */}
            <div>
                <NavBar/>
                <Switch>
                    <Route exact path='/' component={HomeScreen} />
                    <Route exact path='/about' component={AboutScreen} />
                    <Route exact path='/login' component={LoginScreen} />
                    {/* Default page o 404 o usar un redirect  */}
                    {/* <Route component={HomeScreen}/> */}
                    {/* Si no encuentra ninguna de esatas condiciones redirect to / */}
                    <Redirect to='/'/>
                </Switch>
            </div>
        </Router>
    )
}
