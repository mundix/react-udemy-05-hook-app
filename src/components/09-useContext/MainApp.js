import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    // Hay que pasarle el value si no da un error de props , a parentemetne es obligatorio ene sta 
    // Version de react , en value es loq ue queiro compartir por ejemplo el user 

        const [user, setUser] = useState({});

    return (
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <AppRouter />
        </UserContext.Provider>
    )
}
