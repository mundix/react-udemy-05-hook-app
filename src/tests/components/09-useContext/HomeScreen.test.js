import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen/>', () => {
    // const wrapper = shallow(<HomeScreen/>);
    const user = {
        name: 'Edmundo',
        email: 'ce.pichardo@gmail.com'
    }
    // Para renderizar el componente hijo , necesita le  mount instead the shallow 
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen/>
        </UserContext.Provider>
        );
    // Como hago para establecer un contexto 
    
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    

})
