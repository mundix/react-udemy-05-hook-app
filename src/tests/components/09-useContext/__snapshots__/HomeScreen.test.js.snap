// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Edmundo",
   "email": "ce.pichardo@gmail.com"
}
  </pre>
</div>
`;
